src/js/Application.jsx

```diff
@@ -411,14 +420,41 @@ export default class Application extends Component {
     }
 
     let footerStyle = this.state.showFooter ? "footroom-wrapper" : "footroom-wrapper__hide";
+
+    // Determine the headroom space at the top of the scrollable pane (not related to headroom.js)
+    // console.log("Determine the headroom space pathname:" + pathname);
     let appBaseClass = "app-base";
-    if (isCordova()) {
-      if (!hasIPhoneNotch()) {
-        appBaseClass += " cordova-base-iphonex";
+    if (isWebApp()) {
+      appBaseClass += " headroom-webapp";
+    } else {
+      appBaseClass += " cordova-base";
+      if (isIOS()) {
+        appBaseClass += " headroom-ios";
```

Comment:
Hi @SailingSteve, we often search for styles based on the BEM naming conventions that assume stylenames like "app-base-headroom-webapp" are all one continuous script. http://getbem.com/naming/ 
Lines that start with "__" or "--" indicate a style definition that is on its own line in the scss, and style names with "-" in them should not be split across multiple lines. Could you please refactor this pull request?

Reply:
@DaleMcGrew 

Hi Dale,
  I made the change you requested.  Please don't merge it, I haven't tested the new way yet (takes an hour or two to go through all the permutations). 
  We build up class names conditionally all over the place, but the way I did it here takes that further than before.
  To me the continuous way seems like sprawling duplicated code, about 3x as much, and less clear and more mistake prone than before.  I'll probably have to add sizes for iPads at some point, so it will get longer.
  If you really want it this continuous way, I'll test it and remove the "built up" way.

Thanks,
Steve

Reply:
Hi Steve, code like this where there is a space breaking up the styles is fine:
let appBaseClass = "app-base";
if (isWebApp()) {
  appBaseClass += " headroom-webapp";

What would *break* BEM naming standards (and how we search for the styles across our code base) would be if you did this:
let appBaseClass = "app-base";
if (isWebApp()) {
  appBaseClass += "-headroom-webapp";

Or this, since "getAndroidSize" tacks on text like "-xl". You want to keep all styles like "headroom-android-xl" as one continue string. Now if you made "getAndroidSize" return "--xl" then that would be ok and would follow BEM:
appBaseClass += " headroom-android";
appBaseClass += getAndroidSize();

Reply:
My original pass met this criteria: "Lines that start with "__" or "--" indicate a style definition that is on its own line in the scss, and style names with "-" in them should not be split across multiple lines. Could you please refactor this pull request?"  See line 48 through 74 of _bass.scss in this PR.

Reply:
@DaleMcGrew 
I updated all we talked about this morning, and merged in yesterday's changes that eliminate the FacebookAPI exception that was manifesting in the phone when you clicked into a text field.
I'll go on to the twitter signin issue.